### PR TITLE
iboss Sentinel Solution

### PR DESCRIPTION
**Pull request by mvforgione**
Required items, please complete
   
   Change(s):
   - Initial commits

   Reason for Change(s):
   - New iboss solution for Microsoft Sentinel

   Testing Completed:
   - Passed!  - Failed:     0, Passed:  1872, Skipped:     0, Total:  1872, Duration: 25 s - Kqlvalidations.Tests.dll (netcoreapp3.1)

### CODE REVIEW

File: Solutions/iboss/Data Connectors/iboss_cef.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      "description": "Set your Threat Console to send Syslog messages in CEF format to your Azure workspace. Make note of your Workspace ID and Primary Key within your Log Analytics Workspace (Select the workspace from the Log Analytics workspaces menu in the Azure portal. Then select Agents management in the Settings section) \n\n>1. Navigate to Reporting & Analytics inside your iboss Console\n\n>2. Select Log Forwarding -> Forward From Reporter\n\n>3. Select Actions -> Add Service\n\n>4. Toggle to Microsoft Sentinel as a Service Type and input your Workspace ID/Primary Key along with other criteria\n\n>5. Wait one to two minutes for the setup to complete\n\n>6. Select your Microsoft Sentinel Service and verify the Sentinel Setup Status is Successful"
+    },
+    {
+      "title": "2. Validate connection",
+      "description": "Open Log Analytics to check if the logs are received using the CommonSecurityLog schema.\n\n>It may take about 20 minutes until the connection streams data to your workspace."
+    }
+  ],
+  "metadata": {
+    "id": "Unique Identifier (GUID) used to identify dependencies and content from solutions or community.",
```
Comment (v-sabiraj) [inline Solutions/iboss/Data Connectors/iboss_cef.json:72]: @mvforgione , please add Unique Identifier (GUID), thanks.

**Review by v-sabiraj — CHANGES_REQUESTED** [review]

**Comment by v-sabiraj** [comment]
Hey @oshezaf , will please once have look for the parser , thanks.

File: Solutions/iboss/Parsers/ibossUrlEvent.txt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                , UrlCategory=DeviceCustomString2
+                , HttpRequestMethod=RequestMethod
+                , HttpUserAgent=RequestClientApplication
+                , FileSHA256=DeviceCustomString3
+                , ThreatName=DeviceCustomString1
+                , MalwareDetected=DeviceCustomNumber1
+                , CNCDetected=DeviceCustomNumber2
+| extend NetworkBytes=SrcBytes+DstBytes
+        , EventTime=todatetime(DeviceCustomDate1)
+| project-away DeviceCustomNumber1Label
```
Comment (oshezaf) [inline Solutions/iboss/Parsers/ibossUrlEvent.txt:46]: A style not project-away accepts wilecards, so this command can be shortened to project-away DeviceCustom*, Flex*

**Review by oshezaf — COMMENTED** [review]
General: I cannot validate the parser itself. This requires a testing environment, understanding of the source and the logs. While I provided some general comments, to ensure quality, parsers, like any code, need to be reviewed by someone who is part of the project team.

File: Solutions/iboss/Workbooks/Images/Logo/iboss_full-logo_2020_vector_white.svg [hunk 3]
```diff
@@ -0,0 +1 @@
...5.242-14.679 40.27 4.683 52.291 16.109 10.002 37.339 7.548 50.03-5.784 11.987-12.594 11.998-31.683-.03-44.205-1.84-1.916-3.904-3.618-6.013-5.553v-25.25c9.723 3.555 17.708 9.02 24.193 16.574 23.318 27.16 13.212 69.78-17.713 86.342-11.818 6.33-25.109 8.13-37.763 6.171z"/><path class="st1" d="M0 57.181h25.231v83.97C12.616 141.15 0 131.415 0 123.18v-66zM25.165 23.778V46.18H0C0 34.98 12.583 23.778 25.165 23.778zM387.254 33.244v22.723s-35.515-.18-50.837-.144c-2.493.006-5.009.082-7.47.43-3.034.427-5.352 2.009-5.787 5.29-.42 3.16.971 5.517 3.848 6.874 1.494.705 3.065 1.268 4.64 1.772 12.489 3.997 24.987 7.961 37.48 11.945 1.58.504 3.16 1.019 4.71 1.608 17.497 6.645 25.293 26.435 16.81 43.116-5.327 10.479-14.9 14.79-25.86 15.043-21.962.508-43.944.151-66.273.151v-22.65s41.317-.054 59.47-.074c2.147-.002 4.296.13 6.365-.837 4.981-2.33 5.671-8.422 1.246-11.69-1.765-1.305-3.843-1.937-5.9-2.595-13.596-4.348-27.184-8.72-40.79-13.034-18.057-5.725-23.525-19.889-19.919-36.936 1.825-8.623 7.499-14.26 15.378-17.775 4.13-1.842 8.52-3.122 13.013-3.155 19.806-.145 39.613-.062 59.876-.062z"/><path class="st2" d="M217.523 28.651c7.603-1.246 15.004-1.199 22.845-.028v63.12h-22.845V28.65z"/></g></switch></svg>
```
[the remark this answers is not in the document]
Comment (mvforgione) [reply Solutions/iboss/Workbooks/Images/Logo/iboss_full-logo_2020_vector_white.svg:1]: @v-sabiraj thank you for the assistance on the images. I checked the guidelines and was not able to find the restrictions/rules for these. I'll work with our graphics team to have these updated ASAP.

File: Solutions/iboss/Parsers/ibossUrlEvent.txt [hunk 4]
```diff
@@ -0,0 +1,64 @@
+// Usage Instructions:
+// Paste below query in log analytics, click on Save button and select as Function from drop down by specifying function name as ibossUrlEvent and specifying Legacy Category.
+// This function maps iboss URL events to Azure Sentinel Information Model (ASIM) (https://docs.microsoft.com/azure/sentinel/normalization).
```
[the remark this answers is not in the document]
Comment (mvforgione) [reply Solutions/iboss/Parsers/ibossUrlEvent.txt:3]: @oshezaf thank you for the detailed review. I patterned the parser from other solutions present in the Solutions directory of the repo, so I apologize the wording here. Does Microsoft prefer that ASIM parsers be used for solutions or is it really up to the developer to determine what is the best fit for their data?

[the remark this answers is not in the document]
Comment (oshezaf) [reply Solutions/iboss/Parsers/ibossUrlEvent.txt:3]: Our goal is certainly to have ASIM parsers in solutions. It significantly increases the value customers get from your solution as it enables your solution to use alot of the Sentinel built-in capabilities. It does take more effort in creating the parser.

Today, most if not all solutions do not support ASIM, and ASIM parsers are delivered as part of the ASIM project.

Now, sticking to ASIM field names does help users, even if it does not provide the full ASIM value, so overall it is a good choice.

**Comment by v-sabiraj** [comment]
Hey @mvforgione , the workbook query is failing, please check and update, thanks.
image

File: Solutions/iboss/Parsers/ibossUrlEvent.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (mvforgione) [reply Solutions/iboss/Parsers/ibossUrlEvent.txt]: Thank you for the feedback here. I have removed the wording in the interim as I explore creating an ARM template. I attempted to map all appropriate fields to the ASIM schema and I successfully ran the schema validator (`ibossUrlEvent | getschema | invoke ASimSchemaTester('<schema>')`), so I believe we are getting closer to that particular target.

File: Solutions/iboss/Workbooks/Images/Logo/iboss_full-logo_2020_vector_black.svg [hunk 5]
```diff
@@ -0,0 +1 @@
...93.033-8.883 1.313-13.013 3.155-7.88 3.516-13.553 9.152-15.378 17.775-3.606 17.047 1.862 31.211 19.918 36.936 13.607 4.314 27.195 8.686 40.791 13.034 2.057.658 4.135 1.29 5.9 2.595 4.425 3.268 3.735 9.36-1.246 11.69-2.07.968-4.218.835-6.364.837-18.154.02-59.47.073-59.47.073v22.651c22.328 0 44.31.357 66.272-.15 10.96-.254 20.533-4.565 25.86-15.044 8.483-16.68.687-36.471-16.81-43.116-1.55-.59-3.13-1.104-4.71-1.608-12.493-3.984-24.991-7.948-37.48-11.945-1.575-.504-3.146-1.067-4.64-1.772-2.877-1.357-4.267-3.714-3.848-6.874.435-3.281 2.753-4.863 5.786-5.29 2.462-.348 4.978-.424 7.47-.43 15.323-.036 50.838.144 50.838.144V33.244c-20.263 0-40.07-.083-59.876.062z"/><path d="M220.133 143.315c-23.666-3.663-45.108-20.474-50.535-45.402-6.444-29.597 13.383-55.858 36.768-63.35V59.57c-1.13.984-2.392 2.052-3.62 3.158-16.922 15.242-14.679 40.27 4.683 52.291 16.109 10.002 37.339 7.548 50.03-5.784 11.987-12.594 11.998-31.683-.03-44.205-1.84-1.916-3.904-3.618-6.013-5.553v-25.25c9.723 3.555 17.708 9.02 24.193 16.574 23.318 27.16 13.212 69.78-17.713 86.342-11.818 6.33-25.109 8.13-37.763 6.171zm-2.61-51.572h22.845v-63.12c-7.84-1.17-15.242-1.218-22.845.028v63.092z" style="fill:#58873c"/></g></switch></svg>
```
[the remark this answers is not in the document]
Comment (mvforgione) [reply Solutions/iboss/Workbooks/Images/Logo/iboss_full-logo_2020_vector_black.svg]: @v-sabiraj I understand the logos still have illegal attributes. I'm taking this back to our design team to provide new. Please standby.

**Review by v-sabiraj — APPROVED** [review]

**Pull request merged**